service/storage_service.cc

```diff
@@ -5053,7 +5053,17 @@ future<raft_topology_cmd_result> storage_service::raft_topology_cmd_handler(raft
                                 version, current_version)));
                     }
 
+                    seastar::timer<seastar::lowres_clock> stall_timer;
+                    auto start = std::chrono::steady_clock::now();
+                    stall_timer.set_callback([&] {
+                        ss._shared_token_metadata.mark_stalled(start);
+                    });
```

Comment:
Why do we need a timer? Record the time is was constructed (or made current), and an expiration time, and compare the two during destruction.